src/platform/packages/shared/kbn-workflows/server/repositories/workflow_repository.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { ElasticsearchClient, Logger } from '@kbn/core/server';
+import type { EsWorkflow } from '../..';
+
+export interface WorkflowRepositoryOptions {
+  esClient: ElasticsearchClient;
+  logger: Logger;
+  indexName: string;
+}
+
+export class WorkflowRepository {
```

Comment:
If this class is only used by the `workflow_execution_engine` plugin, won't it be better to move it there?

Reply:
My idea is to replace server's local implementation with this shared one. To have single Workflow storage logic for both plugins